### PR TITLE
added laravel 9 support

### PR DESCRIPTION
**Pull request by rcerljenko**
## Summary
- Added Laravel 9 Support to `composer.json` and tests

### CODE REVIEW

**Review by barryvdh — APPROVED** [review]

File: .github/workflows/run-integration-tests.yml
Comment (mfn) [inline .github/workflows/run-integration-tests.yml:79]: Laravel 9 does not support PHP < 8, so those combinations need to be excluded from the matrix

File: .github/workflows/run-tests.yml
Comment (mfn) [inline .github/workflows/run-tests.yml:24]: Same as above

**Review by mfn — CHANGES_REQUESTED** [review]
Some combinations of PHP / Laravel are not valid in the test matrix, can you please fix them?

File: .github/workflows/run-tests.yml
Comment (rcerljenko) [reply .github/workflows/run-tests.yml:24]: yes i see because laravel 9 is only php8+ compatible but i'm not that much familiar with github actions to split php matrix... can you help?

Comment (mfn) [reply .github/workflows/run-tests.yml:24]: You've to add an `exclude` key, see https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-excluding-configurations-from-a-matrix for an example

Comment (rcerljenko) [reply .github/workflows/run-tests.yml:24]: thx! ok i've updated the PR

File: .github/workflows/run-integration-tests.yml
Comment (mfn) [inline .github/workflows/run-integration-tests.yml:81]: The matrix still generates jobs for PHP 7.4 and Laravel 9, see https://github.com/barryvdh/laravel-ide-helper/actions/runs/1692091626

Maybe the `[…, …]` is not supported and you've to manually list each?
```
- php 7.3
  laravel: 9.*
- php 7.4
  laravel: 9.*
```
Just a guess though

**Review by mfn — CHANGES_REQUESTED** [review]
We're not yet there, the matrix isn't correctly generated yet, see my inline feedback.

Another thing: L9 is not yet available for the integration tests, see https://github.com/barryvdh/laravel-ide-helper/runs/4802033128?check_suite_focus=true#step:4:13

So until this works, we can't merge and probably have to wait until the actual release.

File: .github/workflows/run-integration-tests.yml
Comment (rcerljenko) [reply .github/workflows/run-integration-tests.yml]: ok i've updated matrix to suggested syntax.. sorry it's my first time with github actions syntax

Comment (mfn) [reply .github/workflows/run-integration-tests.yml]: np!

yeah it can be a PITA, also since you're first time contributor, we've to manually enabled running the workflows everytime 😅 (I just did, let's wait for the result)

Comment (mfn) [reply .github/workflows/run-integration-tests.yml]: We're one step further, next problem: orchestra/benchmark also needs to have adapted version constraints, see https://github.com/barryvdh/laravel-ide-helper/runs/4803289512?check_suite_focus=true#step:5:20

I see they have not yet made a new 7.* release at https://github.com/orchestral/testbench-core but their branch is also updated. Maybe we can use `dev-master` there or we need to wait.

FTR: If you feel overwhelmed / frustrated, don't be: I can take it from here later if you want.

Comment (rcerljenko) [reply .github/workflows/run-integration-tests.yml]: @mfn if it's not a problem please take it from here because i don't wan't to break something... thx for the support!  :)

**Comment by mfn** [comment]
@barryvdh nice, thx; seems to work already

Now we've to wait for the composer laravel installer to switch to L9 I guess?

**Comment by barryvdh** [comment]
Yeah let's skip the integration test for Laravel 9 for now, but do just the unit tests.

File: .github/workflows/run-integration-tests.yml
Comment (barryvdh) [reply .github/workflows/run-integration-tests.yml]: Let's wait for the 9.x release to run the ingration tests.

**Comment by barryvdh** [comment]
The tests run :)

**Comment by rcerljenko** [comment]
@barryvdh then it's safe to merge?  :)

**Comment by barryvdh** [comment]
Yeah if @mfn approves

**Review by mfn — APPROVED** [review]
Nice work all, thanks! 🙏🏼 

@barryvdh can you add this to the changelog? I can't push unfortunately:
`- Added Laravel 9 support [#1297 / rcerljenko](https://github.com/barryvdh/laravel-ide-helper/pull/1297)`

**Pull request merged**
